Removed t3_int isr

damosvil opened the pull request:
Removed t3 interrupt service routine to avoid wasting necessary CPU cycles

To be merged after https://github.com/bird-sanctuary/bluejay/tree/feature/dynamic_pwm

damosvil committed:
Removed dithering

damosvil committed:
Dynamic PWM initial commit

Commit:
Fixed center/edge aligned selection and pmw freq selection

damosvil committed:
Spend 1 byte to save 2 cycles in t1_int

damosvil committed:
Updated scripts and legacy EEPROM slots

damosvil committed:
Removed t3_int isr

stylesuxx commented on src/Modules/Isrs.asm:
```diff
@@ -609,22 +609,8 @@ t2_int_exit:
 ;
 ; Timer3 interrupt routine
 ;
```
The task of timer 3 is still commutation timing though - I think we can keep the "Tasks" documentation here, no?

damosvil replied:
It is explained at Bluejay.asm line 43

stylesuxx commented on src/Modules/Timing.asm:
```diff
@@ -691,8 +683,6 @@ comp_read_wrong_startup_jump:
     sjmp comp_check_timeout             ; Continue to look for good ones
 
 comp_read_wrong_extend_timeout:
-    clr  Flag_Demag_Detected            ; Clear demag detected flag
```
Why is it no longer necessary to clear the demag detection flag?

damosvil replied:
Good catch

damosvil replied:
Fixed

damosvil replied:
Fixed

Commit:
Restored instruction removed by mistake

damosvil committed:
Restored removed comment

stylesuxx reviewed (approved):
Looking good to me!

Let's merge the other PR first and I'll merge this one into the v0.21 branch afterwards.

damosvil committed:
Fixed times to be loaded inmediately in Timer3.
In the previous t3 isr version when t3 expires it reloads the timer after a small delay.
In this version we have to load the reload times directly into timer3 and also update the reload time to have a functionality as close as possible to previous version's one.

damosvil committed:
Restored interrupt enable disable previous version functionality

damosvil committed:
Put all Timer3 configurations in critical sections

Commit:
Reduced Timer3 handling critical sections length to avoid communications problems because not processing interrups on time
